efi/bootmanager_policy.go

```diff
@@ -175,13 +175,16 @@ func AddBootManagerProfile(profile *secboot_tpm2.PCRProtectionProfile, params *B
 		}
 	}
 
-	// Iterate over all of the branch points starting from the root and creates a tree of
-	// sub-profiles with AddProfileOR. The ordering doesn't matter here, because each subprofile
-	// is already complete
-	for _, b := range allBranches {
+	// Iterate over all of the branch points starting from the leaves and creates a tree of
+	// sub-profiles with AddProfileOR.
+	for i := len(allBranches) - 1; i >= 0; i-- {
```

Comment:
the reason for this change is not super obvious as the comment doesn't see anything one way or another about order, but the for loops definitely picks one

Reply:
It's because the API is bit stricter with regards to ordering now. That said, this loop is going to disappear in a follow up PR.